### PR TITLE
Step 1: Make code compile well.

### PR DESCRIPTION
**Pull request by reyoung**
I think this PR should be done step by step.

1. Make compile pass.
1. Make all unit tests pass.
1. Daily Merge Develop Branch

Since our refactoring unit-test coverage is good. If we make unit tests pass, we are confident that our code is correct.

This PR just make compile pass for most our code, except.

* [ ] grad_op_builder.cc
* [ ] backward_test.cc
* [ ] rnn_test.cc

### CODE REVIEW

File: paddle/framework/framework.proto [hunk 1]
```diff
@@ -40,8 +40,8 @@ message OpDesc {
   };
 
   message Var {
-    required string name;                 // e.g. "X"
-    optional int dup = 2 [ default = 0 ]; // e.g., "1"
+    required string op_proto_name = 1;
```
Comment (wangkuiyi) [inline paddle/framework/framework.proto:43]: Would `global_name` a shorter and the same accurate name for `op_proto_name`?

File: paddle/framework/framework.proto [hunk 2]
```diff
@@ -40,8 +40,8 @@ message OpDesc {
...
+    repeated string var_names = 2;
```
Comment (wangkuiyi) [inline paddle/framework/framework.proto]: How about `local_names` means Operator-local names?

I am not sure that I understand these two fields; we need some comments here.

**Review by wangkuiyi — APPROVED** [review]
LGTM!

File: paddle/framework/framework.proto (hunk 1 above)
Comment (reyoung) [reply paddle/framework/framework.proto:43]: It is not `global`/`local` relationship.
The relationship of them should be similar with function's parameter and argument.
Like,

```python
fc_layer(X=["x1", "x2"], W=["w1", "w2"])
```

`X` is op_proto_name, `[x1, x2]` is var_names.

Comment (wangkuiyi) [reply paddle/framework/framework.proto:43]: Agreed -- they are like parameters and argument. How about we name them in this way?

Comment (reyoung) [reply paddle/framework/framework.proto:43]: So maybe just like a programming language.

* `op_proto_name` should be changed to `parameter_name`;
* `var_names` should be changed to `argument_names`;

Comment (wangkuiyi) [reply paddle/framework/framework.proto:43]: Cool! Or parameter_names => parameters, argument_names => arguments

**Pull request merged**